lib/encoding/yaml/yaml.toit

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2023 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import binary show LITTLE-ENDIAN
+import bitmap
+import reader show Reader BufferedReader
+import .encoder
+import .parser
+
+export YamlEncoder
+
+INITIAL-BUFFER-SIZE_ ::= 64
+MAX-BUFFER-GROWTH_ ::= 1024
+
+/**
+Encodes the $obj as a YANL ByteArray.
+The $obj must be a supported type, which means either a type supported
+  by the $converter block or an instance of int, bool, float, string, List
+  or Map.
+Maps must have only string keys.  The elements of lists and the values of
+  maps can be any of the above supported types.
+The $converter block is passed an object to be serialized and an instance
+  of the $YamlEncoder class.  If it returns a non-null value, that value will
+  be serialized instead of the object that was passed in.  Alternatively,
+  the $converter block can call the $YamlEncoder.encode, $YamlEncoder.put-list,
+  or $YamlEncoder.put_unquoted methods on the encoder.
+Utf-8 encoding is used for strings.
+*/
+encode obj [converter] -> ByteArray:
+  e := YamlEncoder
+  e.encode obj converter
+  return e.to-byte-array
+
+encode obj converter/Lambda -> ByteArray:
```

Comment:
```suggestion
/**
Variant of $(encode obj [converter]).
Takes a lambda instead of a block as $converter.
*/
encode obj converter/Lambda -> ByteArray:
```